### ISSUE TITLE
Multiple dex files define Landroid/support/annotation/AnimRes (Cordova 5.0.0 / Cordova Android 4.0.0)

### ISSUE THREAD

**Issue by BlueHayes** [Wizcorp/phonegap-facebook-plugin#997]
After upgrading to Cordova 5.0.0 and Cordova Android 4.0.0, I can no longer compile apps with phonegap-facebook-plugin installed. I know very little about native Android development and the associated build steps, I've tried my best to find a solution to this for many hours now but haven't been able to resolve it.

```
  	UNEXPECTED TOP-LEVEL EXCEPTION:
  	com.android.dex.DexException: Multiple dex files define Landroid/support/annotation/AnimRes;
  		at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:596)
  		at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:554)
  		at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:535)
  		at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:171)
  		at com.android.dx.merge.DexMerger.merge(DexMerger.java:189)
  		at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:454)
  		at com.android.dx.command.dexer.Main.runMonoDex(Main.java:303)
  		at com.android.dx.command.dexer.Main.run(Main.java:246)
  		at com.android.dx.command.dexer.Main.main(Main.java:215)
  		at com.android.dx.command.Main.main(Main.java:106)
```

Thanks

**Comment by deliverymanager** [comment]
I had the same issue.
It is due to the fact that the Facebook plugin and the push plugin use a different version of android jar file.
Take a look at the recent issues and you will find an answer there.

**Comment by deliverymanager** [comment]
https://github.com/Wizcorp/phonegap-facebook-plugin/issues/992

**Comment by BlueHayes** [comment]
The issues seem to be a tad different (although probably stemming from a single root cause). I don't use com.phonegap.plugins.PushPlugin and yours relates to a "Landroid/support/v4/accessibilityservice/AccessibilityServiceInfoCompat$AccessibilityServiceInfoVersionImpl" definition rather than "Landroid/support/annotation/AnimRes".

**Comment by aogilvie** [comment]
You have conflicting support files. See this fix: http://stackoverflow.com/a/26713987

**Comment by MingxuanLi** [comment]
Did anyone resolve this already ? Can you please advise the solution ?

**Comment by mabdelfattah** [comment]
I've this problem but I don't use Push plugin!

Here are my plugins list (in addition to phonegap-facebook-plugin):

    com.ionic.keyboard 1.0.4 "Keyboard"
    com.phonegap.plugins.barcodescanner 2.0.1 "BarcodeScanner"
    com.pushwoosh.plugins.pushwoosh 3.5.4 "Pushwoosh"
    cordova-plugin-device 1.0.1 "Device"
    cordova-plugin-dialogs 1.1.1 "Notification"
    cordova-plugin-geolocation 1.0.1 "Geolocation"
    cordova-plugin-globalization 1.0.1 "Globalization"
    cordova-plugin-google-analytics 0.8.0 "Google Universal Analytics Plugin"
    cordova-plugin-inappbrowser 1.0.1 "InAppBrowser"
    cordova-plugin-network-information 1.0.1 "Network Information"
    cordova-plugin-splashscreen 2.1.0 "Splashscreen"
    cordova-plugin-statusbar 1.0.1 "StatusBar"
    cordova-plugin-whitelist 1.0.0 "Whitelist"
    nl.x-services.plugins.launchmyapp 3.2.4 "Custom URL scheme"
    nl.x-services.plugins.socialsharing 4.3.8 "SocialSharing"
    org.pushandplay.cordova.apprate 1.1.7 "AppRate"

**Comment by MingxuanLi** [comment]
@mabdelfattah 

The Google Analytics Plugin might clash with Facebook plugin...

**Comment by MingxuanLi** [comment]
@aogilvie 

We still encounter this problem, can you advise the steps to fix this ? It looks like Facebook plugin still using the old android-support-v4.jar.

**Comment by diego-link-eggy** [comment]
http://stackoverflow.com/questions/31451872/cordova-com-danielcwilson-plugins-googleanalytics-with-com-admob-google/31498952?noredirect=1#comment52124000_31498952

**Comment by BlueHayes** [comment]
The fix at the bottom of #1090 works for me (should of course be `build-extras.gradle` rather than the typo). Seems like a bit of a hack - I'm not entirely sure what it's doing - but it works for me.
